### ISSUE TITLE
CuDNNLSTM and activations

### ISSUE THREAD

**Issue by clemej** [keras-team/keras#8510]
This is likely a silly question, but how are activations meant to be handled between CuDNNLSTM/CuDNNGRU layers? The new CuDNNLSTM and CuDNNGRU layers do not contain an activation= parameter to insert activations between GRU cells, and there is no documentation on how activations interact with these new layers. Should they now be created as separate activation layers? 

Or does the optimization/speedup of these constructs come from not having activations between the layers?   Pointers to documentation I may have missed are always appreciated.   

Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [X] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [X] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by bzamecnik** [comment]
According to CuDNN docs the final `activation` is `tanh` and gate activations (`recurrent_activation` in Keras terminology) are `sigmoid`. Note that default `recurrent_activation` for plain GRU implementation in Keras is `hard_sigmoid`.

I agree that it would be good to improve documentation of Keras and mention activations (even if they're not configurable) and possibly provide links to related CuDNN or TensorFlow documentation.

Speedup in CuDNN implementations of LSTM/GRU is due to a lack of activation, rather in a few clever tricks to use the better utilize the underlying hardware. A detailed description is given in a post in NVIDIA blog [Optimizing Recurrent Neural Networks in cuDNN 5
](https://devblogs.nvidia.com/parallelforall/optimizing-recurrent-neural-networks-cudnn-5/). Note that CuDNN uses a bit different definition of GRU than is common.

**Comment by zixia** [comment]
Does the following default activitations right?

- `tf.keras.layers.CuDNNGRU`: `hard_sigmoid`
- `tf.contrib.cudnn_rnn.CudnnLSTM`: `tanh`

I ran into this issue today and I believe we should put them into the documents.

**Comment by gabrieldemarmiesse** [comment]
PR on the documentation welcome.

**Comment by Sayan98** [comment]
Hey, @gabrieldemarmiesse I would like to take up this issue.

**Comment by k--chow** [comment]
Correct me If I'm wrong but nobody has updated the documentation yet. May I take on this issue?

**Comment by lordsoffallen** [comment]
I believe cuDNN backend provised some other stuff that maybe useful. I wanted to do batch normalization before activations but not possible with GPU. 
[cuDNN Batch Normalization Mode](https://docs.nvidia.com/deeplearning/sdk/cudnn-developer-guide/index.html#cudnnBatchNormMode_t). Any issue that this will resolve in the future?

**Comment by IIIIIIIIIIIIIIIIIIIIIIII-IIIIIII** [comment]
IIIIIIIIIIIIIIIIIIIIIIII IIIIIII 
IIIIIII IIIIIIIIIIIIIIIIIIIIIIII
365

**Comment by IIIIIIIIIIIIIIIIIIIIIIII-IIIIIII** [comment]
> 2025Users Blocked 
By Blank 
Unknown 
Unlimited 
Until Nov 2nd 2020 `Update `

**Comment by IIIIIIIIIIIIIIIIIIIIIIII-IIIIIII** [comment]
Preview 
Blocked Until Tomorrow By 
Billing.isisisters 

Blocking 
Blocking 
Blocking 
Blocking 
Blocking 
Blocking 
I-Invoke.
com

**Comment by IIIIIIIIIIIIIIIIIIIIIIII-IIIIIII** [comment]
`People 
Blocked 
Blogs
Blocked 
Websites 
Blocked
Hackers 
Blocked 
Covid-19 
Blocked
Debugging 
`
